src/datetime_conversions.py

```diff
@@ -0,0 +1,42 @@
+import re
+from datetime import datetime
+
+
+def datetime_to_iso_string(date: datetime) -> str:
+    """
+    Return an ISO-8601 representation of a datetime object.
+    """
+    return date.isoformat() + "Z"
+
+
+# Matches an iso 8601 date representation
+DATE_ISO_REGEX = re.compile(
+    r"(\d\d\d\d)-(\d\d)-(\d\d)T(\d\d):(\d\d):(\d\d)Z", re.UNICODE
+)
+
+# Groups for retrieving values from DATE_ISO_REGEX matches.
+DATE_ISO_YEAR_GRP = 1
+DATE_ISO_MONTH_GRP = 2
+DATE_ISO_DAY_GRP = 3
+DATE_ISO_HOUR_GRP = 4
+DATE_ISO_MIN_GRP = 5
+DATE_ISO_SEC_GRP = 6
+
+
+def iso_string_to_datetime(input_string: str) -> datetime:
+    """
+    Return a datetime object from an iso 8601 representation.
+    Raise ValueError if string is nonconforming.
+    """
+    match = DATE_ISO_REGEX.match(input_string)
+    if not match:
+        raise ValueError(f"String is not a valid iso 8601 datetime: {input_string}")
+    date = datetime(
+        year=int(match.group(DATE_ISO_YEAR_GRP)),
+        month=int(match.group(DATE_ISO_MONTH_GRP)),
+        day=int(match.group(DATE_ISO_DAY_GRP)),
+        hour=int(match.group(DATE_ISO_HOUR_GRP)),
+        second=int(match.group(DATE_ISO_SEC_GRP)),
+        minute=int(match.group(DATE_ISO_MIN_GRP)),
+    )
+    return date
```

Comment:
Already extracted in the tag-value PR:
https://github.com/spdx/tools-python/pull/395/files#diff-aff89e705afafa1b570e0e9c29668a2ed645dfc87b0377fae2dc0f82eb7bec49

Reply:
I just grabbed this from git history. I see there is a different (potentially better) implementation in https://github.com/spdx/tools-python/pull/396/files#diff-aff89e705afafa1b570e0e9c29668a2ed645dfc87b0377fae2dc0f82eb7bec49. But it also raises `SPDXParsingError`, which doesn't make sense in my context. As the other PR is not merged yet, I'd leave this as is for now, and it can be refactored in that PR.
(Of course, if that PR gets merged first, I can rebase and change it here...)